### ISSUE TITLE
Accessing subtoolchain versions in easyconfig file

### ISSUE THREAD

**Issue by akesandgren** [hpcugent/easybuild-framework#2079]
It would sometimes be good to be able to pick out version number of subtoolchains in an easyconfig.
For instance, Paraver-4.5.6-foss-2015a.eb contains:

```toolchain = {'name': 'foss', 'version': '2015a'}

compname = 'GCC'
compver = '4.9.2'

dependencies = [
    ('zlib', '1.2.8', '', (compname, compver)),
    ('wxPropertyGrid', '1.4.15', "", (compname, compver)),
    ('Boost', '1.58.0', '-serial', (compname, compver)),
]```

It would be better to be able to pick out the version of GCC used in the active toolchain instead of hard coding it to 4.9.2. 
Something like:
```
gccver = getver('GCC')
```
or similar.
And it is not just GCC that could be of interest...

**Comment by ocaisa** [comment]
Hmm, minimal toolchains already covers most of this requirement, it inserts the appropriate toolchain and version based on the available easyconfigs. There should be enough in that method so that you could handle the dependency syntax
```
('zlib', '1.2.8', '', toolchain_name),
```
and never have to give a version number again...except in toolchain definitions.

**Comment by akesandgren** [comment]
Yes, it might just be that i was hitting some badly written easyconfigs... Just back from a 5 week vacation so I'm not up to speed yet :-)

It's sort of related to the %(pyver) macro. I'll see if i find more concrete examples where it would be useful that isn't covered already.

**Comment by ocaisa** [comment]
Whoops, I wasn't clear. What I meant was that the machinery to make this possible is there...but the implementation is not!

**Comment by akesandgren** [comment]
Ah, yeah, Kenneth told me as much. Just leave it for the time being. I'll see what i run into...
